src/EditorFeatures/CSharpTest/ImplementAbstractClass/ImplementAbstractClassTests.cs

```diff
@@ -1818,6 +1818,151 @@ public override void M<T>()
     {
         throw new System.NotImplementedException();
     }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsImplementAbstractClass)]
+        public async Task TestAllowNullAttributeOnNonNullableReferenceType()
+        {
+            await TestInRegularAndScript1Async(
+@"#nullable enable
+
+abstract class Base<T>
+{
+    [return: System.Diagnostics.CodeAnalysis.MaybeNullAttribute]
+    protected abstract T Method([System.Diagnostics.CodeAnalysis.AllowNullAttribute] T value);
+}
+
+class [|Class|] : Base<string>
+{
+#pragma warning disable CS8609 // Nullability of reference types in return type doesn't match overridden member.
+}",
+@"#nullable enable
+
+abstract class Base<T>
+{
+    [return: System.Diagnostics.CodeAnalysis.MaybeNullAttribute]
+    protected abstract T Method([System.Diagnostics.CodeAnalysis.AllowNullAttribute] T value);
+}
+
+class Class : Base<string>
+{
+#pragma warning disable CS8609 // Nullability of reference types in return type doesn't match overridden member.
```

Comment:
For cases where we are generating a non null annotated type implementation I believe we should keep it non null. It still fulfills the condition of "MaybeNull" but the generated implementation would encourage not null.

Reply:
Specifically for return type. If the attribute allows null for parameters we should respect that.

Reply:
> It still fulfills the condition of "MaybeNull"

Then we need to preserve the attribute to support flow analysis.

Reply:
Is that true? If the interface has the attributes and you use that, it should be respected in flow analysis. If you're using an implementation, it would depend on the type anyways. 

```
class  C : Base<string>
{
    protected override string Method(string? value) => throw new System.NotImplementedException(); // This override is still allowed right? 
}

class D 
{
    void M(C c)
    {
        c.Method(null).Length; // <-- flow analysis should know that this will not be null
    }
}

Reply:
You are right. I thought that you want to keep the "MaybeNull" semantics.

Reply:
Ping on this @NextTurn . Once this issue is resolved I believe we can review again and prepare for merge